### PR TITLE
xpmem: Added RHEL 8.3 support

### PR DESCRIPTION
**Pull request by amd-toolchain-support**
- Fixed SHA256 value for version 2.6.5 (merged on: 21-Nov-2020)
- Added RHEL 8.3 support for kernel 4.18.0-240.1.1.el8_3.x86_64

### CODE REVIEW

**Comment by haampie** [comment]
Can you turn that patch into a PR for xpmem too?

**Review by haampie — APPROVED** [review]

**Review by haampie — CHANGES_REQUESTED** [review]
Actually no, you shouldn't change the commit for version 2.6.5, your patch should apply to `master` or you should add a new version like `2.6.5-1` for commit `0d0bad4e1d07b38d53ecc8f20786bb1328c446da`.

File: var/spack/repos/builtin/packages/xpmem/package.py [hunk 1]
```diff
@@ -25,7 +25,7 @@ class Xpmem(AutotoolsPackage):
     # Versions starting 2.6.4 are neither tagged nor released in the repo
     # (the choice of commits is based on the commit history of
     # 'kernel/xpmem_private.h'):
-    version('2.6.5', commit='4efeed9cbaabe971f3766d67cb108e2c3316d4b8')
+    version('2.6.5', commit='0d0bad4e1d07b38d53ecc8f20786bb1328c446da')
```
Comment (tldahlgren) [inline var/spack/repos/builtin/packages/xpmem/package.py:28]: Why did the hash need to be changed?

**Comment by amd-toolchain-support** [comment]
> Actually no, you shouldn't change the commit for version 2.6.5, your patch should apply to `master` or you should add a new version like `2.6.5-1` for commit `0d0bad4e1d07b38d53ecc8f20786bb1328c

In present spack downloads below commit id and was merged in the year of 2017.
commit 4efeed9cbaabe971f3766d67cb108e2c3316d4b8
Author: Nathan Hjelm <hjelmn@me.com>
Date:   Mon Aug 28 23:10:08 2017
This commit id was the first release version of 2.6.5.  (You can find version  here “kernel/xpmem_private.h:#define XPMEM_CURRENT_VERSION_STRING     "2.6.5")

The one we proposed and the newer commit id and was merged on Nov 2020. 
commit 0d0bad4e1d07b38d53ecc8f20786bb1328c446da (HEAD -> master, origin/master, origin/HEAD)
Merge: cae8601 2e2fa69
Author: Nathan Hjelm <hjelmn@google.com>
Date:   Sat Nov 21 08:45:11 2020 -0700
Versions can be found here 
dkms.conf:PACKAGE_VERSION="2.6.5"
xpmem-kmod.spec:Version: 2.6.5

This commit id was the 7th revision of version 2.6.5 and merged in Nov 2020. And XPMEM authors havent changed version numbers but updated their source codes. Existing spack recipe with commit id 4efeed9cbaabe971f3766d67cb108e2c3316d4b8 isn’t compatible with many recent Operating Systems.  So we have used latest commit id and created spack recipe with the same version number. 

As per previous communication  you have asked us to change the version number to 2.6.5-1. Please suggest if you want us to change our new recipe to 2.6.5.-1 version, then we will change.

**Comment by haampie** [comment]
Pinging @alalazo / @adamjstewart for an opinion. The situation is that both `2.6.4` and `2.6.5` were never tagged (I"ve created an issue https://github.com/hjelmn/xpmem/issues/41), and our 2.6.4 and 2.6.5 refer to the first commits that changed the version numbers in the source code. The patch provided by @amd-toolchain-support refers to the latest commit on `master`. Should we add our own `2.6.5-1` version for a recent commit so that it's easier to apply this patch? And should we set `preferred=True` on 2.6.5?

**Comment by alalazo** [comment]
> The patch provided by @amd-toolchain-support refers to the latest commit on master

I'd really rather avoid patching moving versions, since the patch is bound to be outdated as the development of the software progresses.

File: var/spack/repos/builtin/packages/xpmem/package.py (hunk 1 above)
Comment (haampie) [inline var/spack/repos/builtin/packages/xpmem/package.py]:
```suggestion
    version('2.6.5-36', commit='0d0bad4e1d07b38d53ecc8f20786bb1328c446da')
    version('2.6.5', commit='4efeed9cbaabe971f3766d67cb108e2c3316d4b8')
```

File: var/spack/repos/builtin/packages/xpmem/package.py [hunk 2]
```diff
@@ -35,6 +35,9 @@ class Xpmem(AutotoolsPackage):
     variant('kernel-module', default=True,
             description='Enable building the kernel module')
 
+    # Added RHEL 8.3 kernel support
+    patch('xpmem_v2.6.5.patch', when="@2.6.5", level=1)
```
Comment (haampie) [inline var/spack/repos/builtin/packages/xpmem/package.py]:
```suggestion
    patch('xpmem_v2.6.5.patch', when="@2.6.5-36", level=1)
```

**Comment by haampie** [comment]
My proposal is to use `[version]-[short git sha]` instead of `[version]-1` since numbering from 1 starting at an arbitrary commit doesn't make much sense either.

**Comment by adamjstewart** [comment]
> numbering from 1 starting at an arbitrary commit doesn't make much sense either.

It makes less sense, although it does allow version sorting to work. As long as this is the only random commit we add that doesn't matter though.

**Comment by haampie** [comment]
```
$ git describe --tags 4efeed9cbaabe971f3766d67cb108e2c3316d4b8 # spack version 2.6.5
v2.6.3-20-g4efeed9
$ git describe --tags 0d0bad4e1d07b38d53ecc8f20786bb1328c446da # new amd version
v2.6.3-56-g0d0bad4
```

so 36 commits after "2.6.5", then `v2.6.5-36` makes sense, maybe. It allows sorting and adding an older version.

**Comment by amd-toolchain-support** [comment]
> so 36 commits after "2.6.5", then `v2.6.5-36` makes sense, maybe. It allows sorting and adding an older version.

Could you please finalize the naming convention, then we will make required changes.

**Comment by haampie** [comment]
The -36 suffix is fine, you can use that

**Review by haampie — APPROVED** [review]

**Review by alalazo — APPROVED** [review]

**Pull request merged**